cmd/distrogen/templates/project/Makefile

```diff
@@ -0,0 +1,173 @@
+include ./make/common.mk
+include ./make/maintenance.mk
+
+MAKEFLAGS += --no-print-directory
+
+SPEC_FILE = {{ .Spec.Path }}
+
+#############
+# Development
+#############
+
+.PHONY: dev-setup
+dev-setup: install-distrogen workspace setup-hooks
+
+.PHONY: setup-hooks
+setup-hooks:
+	git config core.hooksPath $(PWD)/hooks
+
+.PHONY: precommit
+precommit: misspell lint test-distrogen
+
+.PHONY: presubmit
+presubmit: misspell lint
+
+##########################
+# Updating OTel Components
+##########################
+
+.PHONY: update-google-otel-components
+
+update-otel-components: COMPONENT_DIR := components
+update-otel-components: DISTROGEN_QUERY := $(DISTROGEN_BIN) -spec $(SPEC_FILE) -field
+update-otel-components: export OTEL_VERSION = v$(shell $(DISTROGEN_QUERY) opentelemetry_version)
+update-otel-components: export OTEL_CONTRIB_VERSION = v$(shell $(DISTROGEN_QUERY) opentelemetry_contrib_version)
+update-otel-components: go.work install-distrogen
+	cd $(COMPONENT_DIR) && PATH="$(TOOLS_DIR):${PATH}" $(MAKE) update-components
+
+.PHONY: test-otel-components
+
+test-otel-components: COMPONENT_DIR := components
+test-otel-components: go.work
+	cd $(COMPONENT_DIR) && $(MAKE) test-components
+
+###################
+# Distro Generation
+###################
+
+.PHONY: gen
+gen: distrogen
+	@$(GEN_OTEL)
+
+.PHONY: regen
+regen: $(DISTROGEN_BIN)
+	@$(GEN_OTEL) --force
+
+regen-v: $(DISTROGEN_BIN)
+	@$(GEN_OTEL) --force -v
+
+GEN_OTEL = $(DISTROGEN_BIN) generate --spec $(SPEC_FILE) \
```

Comment:
Doesn't actually change anything but it's easier to read if this is above the gen targets.